### PR TITLE
Minor fixes for Cutout2D

### PR DESCRIPTION
**Pull request by larrybradley**
This fixes a couple of minor issues I discovered.  Please include in `1.1`!

### CODE REVIEW

**Comment by mwcraig** [comment]
Will review later tonight...

**Comment by larrybradley** [comment]
Thanks, @mwcraig!

File: astropy/nddata/utils.py [hunk 1]
```diff
@@ -519,9 +519,10 @@ def __init__(self, data, position, size, wcs=None, mode='trim',
             units of pixels.  ``size`` can also be a
             `~astropy.units.Quantity` object or contain
             `~astropy.units.Quantity` objects.  Such
-            `~astropy.units.Quantity` objects must be in pixel or angular
-            units.  See the ``mode`` keyword for additional details on
-            the final cutout size.
+            `~astropy.units.Quantity` objects must be in pixel or
+            angular units.  If ``size`` contains any floats, they will
```
Comment (mwcraig) [inline astropy/nddata/utils.py]: add a note that it applies only to sizes in pixels? (bold below is just to indicate the change)

 if ``size`` is **in pixels and** contains any floats...`

**Comment by mwcraig** [comment]
@embray @astrofrog -- I've tagged this as 1.1 and it is ready to merge. Didn't do the merge myself in case you want to re-milestone it. The changes are minor:

+ Round size in pixels to nearest integer.
+ Fix conditional comparing units.

File: astropy/nddata/utils.py (hunk 1 above)
Comment (larrybradley) [inline astropy/nddata/utils.py]: The rounding of angular sizes was already in there:
https://github.com/astropy/astropy/pull/4216/files#diff-9d3c0ca15cfa1e9110d749db5d607202R649
Now, `size` is always rounded.

BTW, I came across this issue testing `size = 0.3*u.arcsec / (0.1*u.arsec/u.pix)  [= 2.999999.... pix]`, expecting `3x3` pixel cutouts, but getting `2x2` cutouts.

Comment (mwcraig) [inline astropy/nddata/utils.py]: I think what is confusing is that it currently says that any sizes will be converted to integers, but what you really mean is that `size` will be converted to an integer number of pixels, rounding to the nearest pixel.

When I first read the line last night (and again this morning) I took the current phrasing to mean that a size in angular units would, if it was a float, be rounded to the nearest integer...that may be an idiosyncratic reading, so revise or not as you think is appropriate :)

Comment (larrybradley) [inline astropy/nddata/utils.py]: Ah, sorry I misunderstood!  Yes, you're correct.  I'll add this to the docs:  "`size` will be converted to an integer number of pixels, rounding to the nearest pixel."

Comment (larrybradley) [inline astropy/nddata/utils.py]: Done, thanks!

**Comment by astrofrog** [comment]
Only affects master following #4187 so no need for a changelog entry, and 1.1 is fine.

**Pull request merged**